Sources/Plasma/PubUtilLib/plGLight/plShadowMaster.cpp

```diff
@@ -88,10 +88,11 @@ void plShadowMaster::SetGlobalMaxSize(uint32_t s)
         int i;
         for( i = 31; i >= 0; i-- )
         {
-            if( (1 << i) & s )
+            if ((1u << i) & s)
                 break;
         }
-        s = 1 << i;
+        hsAssert(i >= 0, "Invalid non-POT shadow size");
+        s = 1u << i;
```

Comment:
cppcheck is still warning about `i` potentially being `uint32_t(-1)`, but I think that's because `hsAssert` doesn't guarantee end of execution when `HS_DEBUGGING` is set.